src/diffusion_for_multi_scale_molecular_dynamics/noise_schedulers/variance_sampler.py

```diff
@@ -160,6 +193,38 @@ def _create_epsilon_array(
             ]
         )
 
+    @staticmethod
+    def _create_beta_array(num_time_steps: int) -> torch.Tensor:
+        return 1.0 / (num_time_steps - torch.arange(1, num_time_steps + 1) + 1)
+
+    @staticmethod
+    def _create_alpha_bar_array(beta_array: torch.Tensor) -> torch.Tensor:
+        return torch.cumprod(1 - beta_array, 0)
+
+    @staticmethod
+    def _create_q_matrix_array(
+        beta_array: torch.Tensor, num_classes: torch.Tensor
+    ) -> torch.Tensor:
+        beta_array_ = beta_array.unsqueeze(-1).unsqueeze(-1)
```

Comment:
I think you inverted `beta` and `1-beta`. 
This is what it should be:
<img width="376" alt="Screenshot 2024-11-01 at 2 49 49 PM" src="https://github.com/user-attachments/assets/2c70b1b5-a03d-45a1-a278-16a75f6cf4c6">

Here is what the code is currently doing:
<img width="401" alt="Screenshot 2024-11-01 at 2 51 31 PM" src="https://github.com/user-attachments/assets/eae8d90f-1b8d-45a1-8e64-23dde5ee4ce8">